File: 6일차/요셉/네트워크.py
def dfs(n, computers, i, visited):
    visited[i] = True
    
    for j in range(n):
        if not visited[j] and computers[i][j] == 1:
            dfs(n, computers, j, visited)

def solution(n, computers):
    answer = 0
    visited = [False for i in range(n)]
    
    for i in range(n):
        if not visited[i]:
            dfs(n, computers, i, visited)
            answer += 1
            
    return answer

File: 6일차/현동/네트워크.js
// https://programmers.co.kr/learn/courses/30/lessons/43162

function dfs(i, n, visited, computers) {
  visited[i] = true;

  for (let j = 0; j < n; j++) {
    if (visited[j] || computers[i][j] === 0) continue;
    dfs(j, n, visited, computers);
  }
}

function solution(n, computers) {
  let answer = 0;
  let visited = Array(n).fill(false);

  for (let i = 0; i < n; i++) {
    if (visited[i]) continue;
    answer++;
    dfs(i, n, visited, computers);
  }

  return answer;
}
